core/src/main/java/io/questdb/griffin/engine/functions/conditional/CaseCommon.java

```diff
@@ -248,8 +252,11 @@ static Function getCaseFunction(int position, int returnType, CaseFunctionPicker
 
         typeEscalationMap.put(Numbers.encodeLowHighInts(BOOLEAN, BOOLEAN), BOOLEAN);
 
-        typeEscalationMap.put(Numbers.encodeLowHighInts(LONG256, LONG256), LONG256);
         typeEscalationMap.put(Numbers.encodeLowHighInts(UUID, UUID), UUID);
+        typeEscalationMap.put(Numbers.encodeLowHighInts(UUID, STRING), UUID);
```

Comment:
Shouldn't this always result in the wider type? UUID and STRING should result in STRING and not UUID. Since every UUID can be casted to STRING, but not every STRING is UUID. What happens when the String is not a UUID?

Reply:
That's a valid point. Let me change the output type to string here and for the varchar.

Reply:
I guess this is also begging for a test.

Reply:
There is a test for that already `CoalesceFunctionFactoryTest#testCoalesceUuidStringLiteral()`

Reply:
After some thinking, I'm not sure if we should do the proposed change. Both of the following `coalesce` calls produce UUID result in Postgres:
```sql
create table x (u uuid);
insert into x values ('00000000-0000-0000-0000-000000000000');
select coalesce(u, 'e8beef38-cd7b-43d8-9b2d-34586f6275fa') from x;
select coalesce('e8beef38-cd7b-43d8-9b2d-34586f6275fa', u) from x;
```

Reply:
I'd rather expand this change to more types with string literals, e.g. IPv4. In those rare cases when users want a string, they can cast UUID/IPv4 to string/varchar explicitly. WDYT?

Reply:
postgres compatibility is a strong argument indeed. 

postgres also produces an error `select coalesce(u, 'not a UUID') from x;`
so I think your PR is good and this can be resolved.